src/Components/test/E2ETestMigration/Tests/BindTest.cs

```diff
@@ -0,0 +1,1367 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using BasicTestApp;
+using Microsoft.AspNetCore.BrowserTesting;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.Testing;
+using PlaywrightSharp;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.Tests
+{
+    public class BindTest : ServerTestBase<ToggleExecutionModeServerFixture<Program>>
+    {
+        public BindTest(
+            ToggleExecutionModeServerFixture<Program> serverFixture,
+            ITestOutputHelper output)
+            : base(serverFixture, output)
+        {
+        }
+
+        protected override Type TestComponent { get; } = typeof(BindCasesComponent);
+
+        protected override async Task InitializeCoreAsync(TestContext context)
+        {
+            await base.InitializeCoreAsync(context);
+
+            // On WebAssembly, page reloads are expensive so skip if possible
+            //Navigate(ServerPathBase, noReload: _serverFixture.ExecutionMode == ExecutionMode.Client);
+        }
+
+        protected async Task<string> GetInputValue(string selector)
+            => await TestPage.EvalOnSelectorAsync<string>(selector, "e => e.value");
+
+        [Theory]
+        [InlineData(BrowserKind.Chromium)]
+        [InlineData(BrowserKind.Firefox)]
+        [InlineData(BrowserKind.Webkit)]
+        // NOTE: BrowserKind argument must be first
+        public async Task CanBindTextbox_InitiallyBlank(BrowserKind browserKind)
+        {
+            if (ShouldSkip(browserKind)) 
+            {
+                return;
```

Comment:
I might be misreading this, but it looks as if any time we choose to skip a given test on a given browser, it would count as a "pass" as far as CI stats are concerned. If so that would be pretty misleading and might make it harder for us to reason about pass rate stats, whether to trigger retries, etc.

Reply:
Yeah this is a bit funky, and worth discussing more, so the full problem is that, we have various helix queues for prs (win 10, ubuntu, osx), there's an even larger set for the full helix matrix that runs once a day.  And then there's 3 kinds of browsers (Chromium/Firefox/Webkit), there's a set of configuration files that specifies which browser and supported for which os.  Right now firefox is busted on ubuntu (so its skipped there, infrastructure related), so I was thinking that generally we will want to mark all tests with all 3 browsers, and this ShouldSkip check is what allows us to no-op the test if the browser is not supported on that OS.  The alternative to this would be getting rid of the BrowserKind attribute/parameter, and having a test per browser (would result in a different factoring of the tests, class or file per browserkind, but still doable if we want clean CI stats per test)